server/aws-lsp-codewhisperer/src/language-server/netTransform/utils.ts

```diff
@@ -0,0 +1,158 @@
+    await copyReferenceDlls(request, basePath)
+    await copySoureFiles(request, basePath)
+    return await zipArtifact(basePath)
+}
+
+export function cleanup(basePath: string) {
+    try {
+        const artifactFolder = path.join(basePath, artifactFolderName)
+        const zipFile = path.join(basePath, zipFileName)
+        fs.rmSync(artifactFolder, { recursive: true, force: true })
+        fs.unlinkSync(zipFile)
+    } catch (error) {
+        console.log('failed to cleanup:' + error)
+    }
+}
+
+export async function createRequirementJson(request: QNetStartTransformRequest, basePath: string) {
+    const fileContent = await createRequirementJsonContent(request)
+    const dir = getRequirementJsonPath(basePath)
+    await writeReuqirmentJsonAsync(dir, JSON.stringify(fileContent))
+}
+
+export async function copyReferenceDlls(request: QNetStartTransformRequest, basePath: string) {
+    const filteredReferences = filterReferences(request)
+    filteredReferences.forEach(reference =>
+        copyFile(reference.AssemblyFullPath, getReferencePathFromRelativePath(basePath, reference.RelativePath))
+    )
+}
+
+export async function copySoureFiles(request: QNetStartTransformRequest, basePath: string) {
+    request.SourceCodeFilePaths.forEach(filePath => {
+        const relativePath = normalizeSourceFileRelativePath(request.SolutionRootPath, filePath)
+        copyFile(filePath, getSourceCodePathFromRelativePath(basePath, relativePath))
+    })
+}
+
+export async function createRequirementJsonContent(request: QNetStartTransformRequest): Promise<RequirementJson> {
+    const projectToReference = request.ProjectMetadata.map(p => {
+        return {
+            project: normalizeSourceFileRelativePath(request.SolutionRootPath, p.ProjectPath),
+            references: p.ExternalReferences.map(r => {
+                return {
+                    AssemblyFullPath: '',
+                    IncludedInArtifact: r.IncludedInArtifact,
+                    ProjectPath: normalizeSourceFileRelativePath(request.SolutionRootPath, r.ProjectPath),
+                    RelativePath: r.RelativePath,
+                    TargetFrameworkId: r.TargetFrameworkId,
+                }
+            }),
+        }
+    })
+    const fileContent: RequirementJson = {
+        ProjectToReference: projectToReference,
+    }
+    console.log('total project reference:' + projectToReference.length)
+    return fileContent
+}
+
+export function filterReferences(request: QNetStartTransformRequest) {
+    //remove duplicate externalreference
+    const externalReferences = request.ProjectMetadata.flatMap(r => r.ExternalReferences)
+    return externalReferences
+        .filter(reference => reference.IncludedInArtifact)
+        .filter(
+            (reference, index) =>
+                index === externalReferences.findIndex(other => reference.AssemblyFullPath === other.AssemblyFullPath)
+        )
+}
+
+export async function zipArtifact(basePath: string): Promise<string> {
```

Comment:
I don't think this will work in a cross platform way. Here is an example in the same package of zipping the workspace in a (mostly) platform-agnostic way. One of the key bits here is we can write the stream directly to S3 instead of going to disk first:

https://github.com/mj632/language-servers/blob/c32400b0345d31e79ef21570ce6f3d1069559638/server/aws-lsp-codewhisperer/src/language-server/dependencyGraph/dependencyGraph.ts#L151

I don't think the workspace zip from that server can be readily reused here (since we have the dependency graph built in, and I don't think we do that for transform), but it might be worth looking and potentially aligning with the filesystem handling there.

Reply:
will do